lib/msal-browser/src/utils/BrowserUtils.ts

```diff
@@ -5,20 +5,38 @@
 import { INetworkModule } from "@azure/msal-common";
 import { FetchClient } from "../network/FetchClient";
 import { XhrClient } from "../network/XhrClient";
+import { BrowserAuthError } from "../error/BrowserAuthError";
 
 /**
  * Utility class for browser specific functions
  */
 export class BrowserUtils {
 
-    // #region Window Navigation
+    // #region Window Navigation and URL management
 
     /**
      * Used to redirect the browser to the STS authorization endpoint
      * @param {string} urlNavigate - URL of the authorization endpoint
+     * @param {boolean} noHistory - boolean flag, uses .replace() instead of .assign() if true
      */
-    static navigateWindow(urlNavigate: string): void {
-        window.location.assign(urlNavigate);
+    static navigateWindow(urlNavigate: string, noHistory?: boolean): void {
+        const windowParentCheck = (window.parent !== window);
+        if (windowParentCheck) {
+            // If we are not in top frame, we shouldn't redirect. This is also handled by the service.
```

Comment:
If this true 100% of the time? We don't want to allow `loginRedirect` and `acquireTokenRedirect` actions inside an iframe, but do we want to always block all redirects inside iframes?

Reply:
We have this code in core for navigateToLoginRequestUrl actions, and the server will also block redirect code requests inside an iframe. We can pass this for the server to throw, or we can block requests. I'm choosing to throw an error here, but we can remove this check if needed.

Reply:
Given this function would be used for legitimate redirects inside an iframe, I would argue the business logic for preventing redirects in certain scenarios should go one level up.

Reply:
done.